File: Core kernel razonamiento repo para Yaiwes/GitHub-Copilot-SDK/GitHub-Copilot-SDK/go/internal/e2e/rpc_ui_ephemeral_query_e2e_test.go
package e2e

import (
	"strings"
	"testing"

	"github.com/github/copilot-sdk/go/internal/e2e/testharness"
	"github.com/github/copilot-sdk/go/rpc"
)

func TestRpcUiEphemeralQuery(t *testing.T) {
	ctx := testharness.NewTestContext(t)
	client := ctx.NewClient()
	t.Cleanup(func() { client.ForceStop() })

	t.Run("should_answer_ephemeral_query", func(t *testing.T) {
		// TODO(cli-1.0.81-2): CLI 1.0.81-5 still fails session.ui.ephemeralQuery against the
		// recorded snapshot on macOS ("Failed to get response from the AI model"). Re-enable
		// once the runtime fix ships.
		t.Skip("blocked on CLI 1.0.81-5 session.ui.ephemeralQuery regression on macOS")

		ctx.ConfigureForTest(t)
		session := createPortedSession(t, client, nil)
		defer session.Disconnect()

		result, err := session.RPC.UI.EphemeralQuery(t.Context(), &rpc.UIEphemeralQueryRequest{
			Question: "In one word, what is the primary color of a clear daytime sky?",
		})
		if err != nil {
			t.Fatalf("UI.EphemeralQuery failed: %v", err)
		}
		if result == nil {
			t.Fatal("Expected non-nil ephemeral query result")
			return
		}
		if strings.TrimSpace(result.Answer) == "" {
			t.Fatal("Expected non-empty ephemeral query answer")
		}
		if !strings.Contains(strings.ToLower(result.Answer), "blue") {
			t.Fatalf("Expected answer to contain blue, got %q", result.Answer)
		}
	})
}

File: ➡️📂Core kernel Yaiwes/GitHub-Copilot-SDK/GitHub-Copilot-SDK/nodejs/test/e2e/rpc_ui_ephemeral_query.e2e.test.ts
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *--------------------------------------------------------------------------------------------*/

import { describe, expect, it } from "vitest";
import { approveAll } from "../../src/index.js";
import { createSdkTestContext, isInProcessTransport } from "./harness/sdkTestContext.js";

describe("UI ephemeral query RPC", async () => {
    const { copilotClient: client } = await createSdkTestContext();

    // TODO(cli-1.0.81-2): CLI 1.0.81-6 still fails session.ui.ephemeralQuery against the
    // recorded snapshot on macOS ("Failed to get response from the AI model").
    it.skipIf(isInProcessTransport || process.platform === "darwin")(
        "should answer ephemeral query",
        { timeout: 120_000 },
        async () => {
            const session = await client.createSession({ onPermissionRequest: approveAll });
            try {
                const result = await session.rpc.ui.ephemeralQuery({
                    question: "In one word, what is the primary color of a clear daytime sky?",
                });

                expect(result).toBeDefined();
                expect(result.answer.trim()).toBeTruthy();
                expect(result.answer.toLowerCase()).toContain("blue");
            } finally {
                await session.disconnect();
            }
        }
    );
});
